### PR TITLE
Solana: script example to read FCD on-chain

### PR DESCRIPTION
**Pull request by gagiuntoli**
The script uses the `caller` program located in `aquarious/programs/caller`

### CODE REVIEW

File: solana-idl/caller.json [hunk 1]
```diff
@@ -162,7 +173,23 @@
...
+          }
+        ]
+      }
+    }
+  ],
   "metadata": {
-    "address": "BmmRtz8Zf4rjQgWT643QG2eqHVkXzebSsnR7XipFTrAg"
+    "address": "BAxZZXMFKHn4cw2NJTM83uLSuuhF3BBq76WLKvDkz5xj"
   }
 }
```
Comment (matheushahnn) [inline solana-idl/caller.json:195]: keep one empty line at the end of the file, please.

File: scripts/solana_read_fcd_onchain.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+const IDL = JSON.parse(fs.readFileSync('./solana-idl/caller.json', 'utf8'));
+
+const main = async () => {
+  const provider = anchor.AnchorProvider.env();
+  anchor.setProvider(provider);
+
+  const callerProgram = new Program(IDL, new PublicKey(callerProgramId), provider);
+
+  const key = 'BTC-USD';
```
Comment (matheushahnn) [inline scripts/solana_read_fcd_onchain.ts:28]: I didn't get what is this key, is it always this value? maybe a setting fits better.

File: scripts/solana_utils.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  const [publicKey] = await derivePDAFromSeed(bufSeed, programId);
+
+  return [publicKey, bufSeed];
+}
+
+export function getKeyPairFromSecretKeyString(secretKey: string): Keypair {
+  return Keypair.fromSecretKey(Uint8Array.from(JSON.parse(secretKey)));
+}
+
+export const prepend0x = (v: string): string => (['0X', '0x'].includes(v.slice(0, 2)) ? v : `0x${v ? v : '0'}`);
```
Comment (matheushahnn) [inline scripts/solana_utils.ts:164]: I'd suggest to use `value` instead of just `v`

**Comment by matheushahnn** [comment]
I would suggest adding some tests

File: scripts/solana_read_fcd_onchain.ts (hunk 2 above)
Comment (gagiuntoli) [reply scripts/solana_read_fcd_onchain.ts:28]: In this case is an example, it should work for any FCD we create on Solana

**Review by DZariusz — APPROVED** [review]
looks ok

**Pull request closed without merging**